contributing/development/handling_compatibility_breakages.rst

```diff
@@ -0,0 +1,168 @@
+.. _doc_handling_compatibility_breakages:
+
+Handling compatibility breakages
+=================================
+
+So you've added a new parameter to a method, changed the return type,
+changed the type of a parameter, or changed its default value,
+and now the automated testing is complaining about compatibility breakages?
+
+Not to worry, there's a system to handle just that!
+
+Compatibility is handled in a few different ways. In GDScript this is simple,
+new arguments are usually given default values when added, so existing scripts
+won't break in such cases.
+
+However in other cases, like GDExtension and C#, the exact way the method is
+declared matters and must be matched exactly. This is handled by creating
+compatibility methods that then get mapped to the new versions of these
+methods. All of this happens *almost* automatically, there are just a few
+things you need to do.
```

Comment:
I feel we need to call out the different types of "compatibility breakages" more explicitly:
- For GDScript, adding a new parameter at the end of a method prototype with a default value does not break compatibility, but it does for GDExtension and C#, as described here. This is the only "compat breakage" that is ok to do in general.
- All other compatibility breakages break GDScript even if compat methods are added, so they can only be done exceptionally with approval for the release management team.

Not sure how to clarify this, with the introduction already being a bit long. I think my main issue is that the light hearted "no worry, there's a system that lets you break compat" may give the wrong impression about what kind of changes are allowed by default.

Reply:
Agreed will see what I can come up with, maybe we can cut this down and make a separate page discussing the general topic of compatibility focused on the abstract instead of the specific